td-shim/src/lib.rs

```diff
@@ -114,8 +114,8 @@ pub struct PayloadInfo {
 }
 
 /// Write three bytes from an integer value into the buffer.
-pub fn write_u24(data: u32, buf: &mut [u8]) {
-    assert!(data < 0xffffff);
+pub fn write_u24(data: u32, buf: &mut [u8; 3]) {
+    assert!(data <= 0xffffff);
```

Comment:
Could you add a unit test for this change?

Reply:
Sure, added.